public/scripts/controllers/challengeForm.js

```diff
@@ -0,0 +1,37 @@
+
+function ChallengeFormCtrl($http, Challenges) {
+  // Challenges available to user
+  this.challenges = Challenges.challenges;
+
+  this.selection = undefined;
+
+  // Handle selection of a challenge
+  this.selectChallenge = function(index) {
+    this.selection = index;
```

Comment:
I believe `this.selection` would be scoped to the function context here.  You probably want to alias `this` before referencing it here.